### DISCUSSION TITLE
Where are the interfaces for the components in v9?

### DISCUSSION THREAD

**Discussion by rexkenley** [microsoft/fluentui#25924] [Q&A] [answered]
Hi 

In v8 each component has an interface, but I can't seem to find it in v9. Where did it go?

Thanks!

**Comment by khmakoto** [comment, marked as answer]
Hi @rexkenley, what do you mean by an interface in this case? Do you mean the props contract?

**Reply by rexkenley** [reply]
Yes, in v8 it has ITextFieldProps for the TextField component for example.

**Reply by khmakoto** [reply]
So, v9 components live in their own packages and get re-exported from the `@fluentui/react-components` package. To find their props contract in our repo you should go to their individual package.

For example, to find the `ButtonProps` for `Button`, you should go to `@fluentui/react-button` and navigate to the `Button.types.ts` file.

https://github.com/microsoft/fluentui/blob/master/packages/react-components/react-button/src/components/Button/Button.types.ts

**Reply by rexkenley** [reply]
Thank you @khmakoto

**Reply by khmakoto** [reply]
No problem!
